In [1]:
import numpy as np
import pandas as pd
import geopandas as gpd
#import geopandas as gpd
import matplotlib.pyplot as plt
import matplotlib.pyplot as plt

### Read datasets

In [2]:

crime17 = pd.read_csv('2017_detroit_crime.csv')
crime13 = pd.read_csv('2013_detroit_crime.csv')






/Users/avigailvantu/anaconda2/envs/myenv/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
/Users/avigailvantu/anaconda2/envs/myenv/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (3) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
print (len(crime13))
print (len(crime17))

146908
82439


In [4]:
crime13[1000:1050]

ROWNUM   CASEID  CRIMEID        CRNO            CATEGORY  \
1000  666662  1619553  1971016  1301030261  MURDER/INFORMATION   
1001  666663  1619552  1971015  1301030260  AGGRAVATED ASSAULT   
1002  666664  1619700  1971213  1301040051      STOLEN VEHICLE   
1003  666665  1620937  1972756  1301070318        SOLICITATION   
1004  666666  1620937  1972756  1301070318  DAMAGE TO PROPERTY   
1005  666667  1764243  2153645  1401030203       MISCELLANEOUS   
1006  666668  1619579  1971045  1301039803  MURDER/INFORMATION   
1007  666669  1619489  1971087  1301030198             LARCENY   
1008  666670  1649327  2008361  1303250240               FRAUD   
1009  666671  1619582  1971048  1301030289             ASSAULT   
1010  666672  1620269  1971921  1301050229       MISCELLANEOUS   
1011  666673  1619580  1971046  1301030287       MISCELLANEOUS   
1012  666674  1619593  1971061  1301030300  DAMAGE TO PROPERTY   
1013  666675  1623555  1976006  1301140278               FRAUD   
1014  666676  1619640  1971119  1301030346            BURGLARY   
1015  666677  1619616  1971086  1301030322  DAMAGE TO PROPERTY   
1016  666678  1619586  1971052  1301030293  AGGRAVATED ASSAULT   
1017  666679  1619590  1971055  1301030297             ASSAULT   
1018  666680  1619578  1971043  1301030286             ASSAULT   
1019  666681  1619587  1971053  1301030294      STOLEN VEHICLE   
1020  666682  1619584  1971050  1301030291       MISCELLANEOUS   
1021  666683  1619585  1971051  1301030292             TRAFFIC   
1022  666684  1764299  2153713  1401030258       MISCELLANEOUS   
1023  666685  1619581  1971047  1301030288       MISCELLANEOUS   
1024  666686  1619629  1971108  1301030335  MURDER/INFORMATION   
1025  666687  1764271  2153677  1401030230       MISCELLANEOUS   
1026  666688  1619600  1971068  1301030307  MURDER/INFORMATION   
1027  666689  1621641  1973639  1301090260  MURDER/INFORMATION   
1028  666690  1620477  1972193  1301060132       MISCELLANEOUS   
1029  666691  1619710  1971225  1301040061       MISCELLANEOUS   
1030  666692  1619687  1971208  1301040038      STOLEN VEHICLE   
1031  666693  1619726  1971244  1301040076      STOLEN VEHICLE   
1032  666694  1619612  1971082  1301030318             ASSAULT   
1033  666695  1619611  1971083  1301030317             ROBBERY   
1034  666696  1619613  1971084  1301030319             LARCENY   
1035  666697  1619603  1971071  1301030310             ASSAULT   
1036  666698  1619520  1971057  1301030228       MISCELLANEOUS   
1037  666699  1619591  1971059  1301030298             LARCENY   
1038  666700  1619725  1971240  1301040075            BURGLARY   
1039  666701  1619648  1971130  1301030354             LARCENY   
1040  666702  1619596  1971064  1301030303             FORGERY   
1041  666703  1619596  1971064  1301030303             LARCENY   
1042  666704  1619604  1971072  1301030311             ROBBERY   
1043  666705  1619899  1971466  1301040239             LARCENY   
1044  666706  1620951  1972773  1301070330       MISCELLANEOUS   
1045  666707  1619622  1971096  1301030328  AGGRAVATED ASSAULT   
1046  666708  1619607  1971077  1301030314             ASSAULT   
1047  666709  1620308  1971967  1301050268             ASSAULT   
1048  666710  1619609  1971080  1301038004  MURDER/INFORMATION   
1049  666711  1619608  1971079  1301030315  MURDER/INFORMATION   

                                     OFFENSEDESCRIPTION  \
1000                                        INFORMATION   
1001        AGG/FEL ASSAULT - NON-FAMILY - OTHER WEAPON   
1002                                      VEHICLE THEFT   
1003               ACCIDENT - FAIL TO STOP AND IDENTIFY   
1004              DAMAGE TO PROPERTY - PRIVATE PROPERTY   
1005                  MISCELLANEOUS - IMPOUNDED VEHICLE   
1006                                        INFORMATION   
1007           LARCENY - PERSONAL PROPERTY FROM VEHICLE   
1008                              FRAUD - IMPERSONATION   
1009             TELEPHONE USED FOR HARASSMENT, T

### Remove NaN's 

In [5]:
#2017
crime17.dropna()

Empty DataFrame
Columns: [Crime ID, Report #, Incident Address, Offense Description, Offense Category, Arrest Charge, Charge Description, State Offense Code, Incident Date & Time, Incident Time (24h), Day of Week (Sunday is 1), Hour of Day, Year, Scout Car Area, Precinct Number, Census Block GEOID, Neighborhood, Council District, Zip Code, Longitude, Latitude, IBR Report Date, Location, uniq]
Index: []

[0 rows x 24 columns]

In [6]:
#2013
crime13.dropna()

ROWNUM   CASEID  CRIMEID         CRNO            CATEGORY  \
0       665662  1629881  1983922   1301310200       MISCELLANEOUS   
1       665663  1618734  1969963   1301010145             LARCENY   
2       665664  1676353  2042431   1305280137       MISCELLANEOUS   
3       665665  1837298  2248340   1407220266               FRAUD   
4       665666  1966686  2416643   1507050157       MISCELLANEOUS   
5       665667  1968282  2418663   1507090164       MISCELLANEOUS   
6       665668  1920783  2358182   1503110144               FRAUD   
7       665669  1900255  2331225   1501060110               FRAUD   
8       665670  1620445  1972151   1301060101  MURDER/INFORMATION   
10      665672  1625450  1978415   1301190500             LARCENY   
11      665673  1632378  1987050   1302070108             LARCENY   
14      665676  1618948  1970242   1301020030  MURDER/INFORMATION   
15      665677  1618852  1970104   1301010262  MURDER/INFORMATION   
16      665678  1618591  1969780   1301010003  AGGRAVATED ASSAULT   
17      665679  1618591  1969780   1301010003             TRAFFIC   
18      665680  1618629  1969833   1301010040  AGGRAVATED ASSAULT   
19      665681  1626234  1979330   1301210242  DAMAGE TO PROPERTY   
20      665682  1619290  1970684   1301030004       MISCELLANEOUS   
21      665683  1623852  1976396   1301150182  DAMAGE TO PROPERTY   
22      665684  1618726  1969951   1301010137  DAMAGE TO PROPERTY   
23      665685  1618595  1969785   1301010007             ASSAULT   
24      665686  1618696  1969918   1301010107             ASSAULT   
25      665687  1618763  1969996   1301010174      STOLEN VEHICLE   
26      665688  1618638  1969855   1301010049      STOLEN VEHICLE   
27      665689  1619881  1971446   1301040221  DAMAGE TO PROPERTY   
28      665690  1618760  1970002   1301010171      STOLEN VEHICLE   
29      665691  1618730  1969956   1301010141  DAMAGE TO PROPERTY   
30      665692  1618702  1969924   1301010113             ASSAULT   
31      665693  1618698  1969920   1301010109             LARCENY   
32      665694  1618601  1969794   1301010013  AGGRAVATED ASSAULT   
...        ...      ...      ...          ...                 ...   
146875  812537  1763577  2152793  1.40101e+09            BURGLARY   
146876  812538  1764218  2153609  1.40103e+09             LARCENY   
146877  812539  1763676  2152919  1.40101e+09      STOLEN VEHICLE   
146879  812541  1763440  2152616  1.31231e+09    WEAPONS OFFENSES   
146880  812542  1763462  2152643  1.40101e+09             LARCENY   
146881  812543  1763452  2152629  1.40101e+09             ASSAULT   
146882  812544  1763486  2152675  1.40101e+09       MISCELLANEOUS   
146883  812545  1764128  2153500  1.40103e+09               FRAUD   
146884  812546  1764128  2153500  1.40103e+09             LARCENY   
146885  812547  1775366  2168004  1.40207e+09               FRAUD   
146886  812548  1763610  2152853  1.40101e+09             LARCENY   
146887  812549  1763621  2152840  1.40101e+09  DAMAGE TO PROPERTY   
146888  812550  1763621  2152840  1.40101e+09             ASSAULT   
146889  812551  1763621  2152840  1.40101e+09             TRAFFIC   
146890  812552  1763675  2152918  1.40101e+09  DAMAGE TO PROPERTY   
146892  812554  1763445  2152627  1.31231e+09     DANGEROUS DRUGS   
146894  812556  1763446  2152623  1.31231e+09    WEAPONS OFFENSES   
146895  812557  1763447  2152630  1.40101e+09     DANGEROUS DRUGS   
146896  812558  1763447  2152630  1.40101e+09    WEAPONS OFFENSES   
146897  812559  1763592  2152806  1.40101e+09            BURGLARY   
146898  812560  1765836  2155638  1.40109e+09             LARCENY   
146899  812561  1763673  2152916  1.40101e+09      STOLEN VEHICLE   
146900  812562  1763556  2152770  1.40101e+09             ASSAULT   
146901  812563  1763467  2152649  1.40101e+09  AGGRAVATED ASSAULT   
146902  812564  1763448  2152625  1.40101e+09  AGGRAVATED ASSAULT   
146903  812565  1763616  2152830  1.40101e+09             ASSAULT   
146904

## Group by Neighborhood

In [7]:
#2017
crime17grouped = crime17.groupby('Neighborhood').count()

In [8]:
#2013
crime13grouped = crime13.groupby('NEIGHBORHOOD').count()

In [9]:
print (len(crime17grouped))
print (len(crime13grouped))

207
157


In [10]:
crime13grouped.head()

ROWNUM  CASEID  CRIMEID  CRNO  CATEGORY  \
NEIGHBORHOOD                                                      
ARDEN PARK/EAST BOSTON     274     274      274   274       274   
ATKINSON-EUCLID            305     305      305   305       305   
BAGLEY                    1730    1730     1730  1730      1730   
BALDUCK PARK              1330    1330     1330  1330      1330   
BARTON MCFARLANE          2694    2694     2694  2694      2694   

                        OFFENSEDESCRIPTION  STATEOFFENSEFILECLASS  \
NEIGHBORHOOD                                                        
ARDEN PARK/EAST BOSTON                 274                    274   
ATKINSON-EUCLID                        305                    305   
BAGLEY                                1730                   1730   
BALDUCK PARK                          1330                   1330   
BARTON MCFARLANE                      2694                   2694   

                        INCIDENTDATE  HOUR   SCA  PRECINCT  COUNCIL  \
NEIGHBORHOOD                                                          
ARDEN PARK/EAST BOSTON           274   274   274       274      274   
ATKINSON-EUCLID                  305   305   305       305      305   
BAGLEY                          1730  1730  1730      1730     1730   
BALDUCK PARK                    1330  1330  1330      1330     1204   
BARTON MCFARLANE                2694  2694  2694      2694     2683   

                        CENSUSTRACT  LOCATION  
NEIGHBORHOOD                                   
ARDEN PARK/EAST BOSTON            7       274  
ATKINSON-EUCLID                 282       305  
BAGLEY                         1730      1730  
BALDUCK PARK                   1330      1330  
BARTON MCFARLANE               2694      2694

In [11]:
#2017
crime17grouped['Neighborhood'] = crime17grouped.index

In [12]:
#2013
crime13grouped['Neighborhood'] = crime13grouped.index

In [13]:
crime13grouped.head(2)

ROWNUM  CASEID  CRIMEID  CRNO  CATEGORY  \
NEIGHBORHOOD                                                      
ARDEN PARK/EAST BOSTON     274     274      274   274       274   
ATKINSON-EUCLID            305     305      305   305       305   

                        OFFENSEDESCRIPTION  STATEOFFENSEFILECLASS  \
NEIGHBORHOOD                                                        
ARDEN PARK/EAST BOSTON                 274                    274   
ATKINSON-EUCLID                        305                    305   

                        INCIDENTDATE  HOUR  SCA  PRECINCT  COUNCIL  \
NEIGHBORHOOD                                                         
ARDEN PARK/EAST BOSTON           274   274  274       274      274   
ATKINSON-EUCLID                  305   305  305       305      305   

                        CENSUSTRACT  LOCATION            Neighborhood  
NEIGHBORHOOD                                                           
ARDEN PARK/EAST BOSTON            7       274  ARDEN PARK/EAST BOSTON  
ATKINSON-EUCLID                 282       305         ATKINSON-EUCLID

In [14]:
d = {'Neighborhood': crime17grouped['Neighborhood'] , 'Incidents17': crime17grouped['Incident Address']}
crime17grouped1 = pd.DataFrame(data=d)

#crime17grouped1['Incident Address']

In [15]:
# create a new DF 
d = {'Neighborhood': crime13grouped['Neighborhood'] , 'Incidents17': crime13grouped['ROWNUM']}
crime13grouped1 = pd.DataFrame(data=d)

In [16]:
crime17grouped1.head(4)

Neighborhood  Incidents17
Neighborhood                           
Airport Sub    Airport Sub          903
Arden Park      Arden Park           27
Aviation Sub  Aviation Sub          254
Bagley              Bagley         1030

In [17]:
crime13grouped1.head(20)

Neighborhood  Incidents17
NEIGHBORHOOD                                               
ARDEN PARK/EAST BOSTON  ARDEN PARK/EAST BOSTON          274
ATKINSON-EUCLID                ATKINSON-EUCLID          305
BAGLEY                                  BAGLEY         1730
BALDUCK PARK                      BALDUCK PARK         1330
BARTON MCFARLANE              BARTON MCFARLANE         2694
BELLE ISLE                          BELLE ISLE          172
BERG-LASHER                        BERG-LASHER          608
BOSTON-EDISON                    BOSTON-EDISON          444
BOYNTON                                BOYNTON          981
BRICKTOWN                            BRICKTOWN          347
BRIGGS                                  BRIGGS          567
BRIGHTMOOR                          BRIGHTMOOR         2062
BRUSH PARK                          BRUSH PARK          436
BURBANK                                BURBANK         2624
CARBON WORKS                      CARBON WORKS          103
CASTLE ROUGE                      CASTLE ROUGE          272
CERVENY                                CERVENY         2227
CHANDLER PARK                    CHANDLER PARK          489
CHENE                                    CHENE          327
CITY AIRPORT                      CITY AIRPORT         3140

In [18]:
crime17

Crime ID    Report #                          Incident Address  \
0       3091097  1707150160       100 block of LAFAYETTE PLAISANCE ST   
1       3010497  1701060130                   1400 block of PARKER ST   
2       3076386  1706120127                   1400 block of PARKER ST   
3       3056009  1704270166                   1400 block of PARKER ST   
4       3076118  1706110180                   1400 block of PARKER ST   
5       3072782  1706030107                   1400 block of PARKER ST   
6       3150044  1711290294             14300 block of GRANDVILLE AVE   
7       3065869  1705180171           Corner of MACK AVE and ALTER RD   
8       3072562  1706020357           Corner of MACK AVE and ALTER RD   
9       3018111  1701260247           Corner of MACK AVE and ALTER RD   
10      3021307  1702040150           Corner of MACK AVE and ALTER RD   
11      3042286  1703250107  Corner of KILBOURNE AVE and DICKERSON ST   
12      3159544  1712260125                 6000 block of WYOMING AVE   
13      3012129  1701110045               12700 block of NASHVILLE ST   
14      3012129  1701110045               12700 block of NASHVILLE ST   
15      3112159  1709010431      Corner of MICHIGAN AVE and MARTIN ST   
16      3112984  1709040049      Corner of MICHIGAN AVE and MARTIN ST   
17      3073476  1706050049      Corner of MICHIGAN AVE and MARTIN ST   
18      3153210  1712070145      Corner of MICHIGAN AVE and MARTIN ST   
19      3011066  1701080029                6400 block of WOODWARD AVE   
20      3079073  1706180076                6400 block of WOODWARD AVE   
21      3146161  1711190045                6400 block of WOODWARD AVE   
22      3146161  1711190045                6400 block of WOODWARD AVE   
23      3029166  1702240189                6400 block of WOODWARD AVE   
24      3045444  1704020100                6400 block of WOODWARD AVE   
25      3045444  1704020100                6400 block of WOODWARD AVE   
26      3009852  1701040232                6400 block of WOODWARD AVE   
27      3018723  1701280010     Corner of FENKELL ST and EVERGREEN RD   
28      3148219  1711250032               11500 block of PETOSKEY AVE   
29      3038936  1703170007     Corner of FENKELL ST and EVERGREEN RD   
...         ...         ...                                       ...   
82409   3077669  1706140438                  15000 block of TROJAN ST   
82410   3157250  1712190135                  15000 block of TROJAN ST   
82411   3137773  1711020307                  15000 block of TROJAN ST   
82412   3023408  1702090372                 15400 block of FENKELL ST   
82413   3026183  1702160270                   5200 block of BISHOP ST   
82414   3016713  1701230226                     2500 block of FORD ST   
82415   3014553  1701170235                     2500 block of FORD ST   
82416   3061854  1705090136                     2500 block of FORD ST   
82417   3012751  1701120245                     2500 block of FORD ST   
82418   3039278  1703170305                     2500 block of FORD ST   
82419   3062042  1705090066                     2500 block of FORD ST   
82420   3024306  1702120058                  15300 block of EDMORE DR   
82421   3055266  1704250442  Corner of W 7 MILE RD and SOUTHFIELD FWY   
82422   3055266  1704250442  Corner of W 7 MILE RD and SOUTHFIELD FWY   
82423   3078919  1706170329  Corner of W 7 MILE RD and SOUTHFIELD FWY   
82424   3088144  1707080238  Corner of W 7 MILE RD and SOUTHFIELD FWY   
82425   3129871  1710140053  Corner of W 7 MILE RD and SOUTHFIELD FWY   
82426   3148021  1711240184  Corner of W 7 MILE RD and SOUTHFIELD FWY   
82427   3154911  1712120057                  7900 block of W OUTER DR   
82428   3014278  1701170033                 14600 block of HUBBELL ST   
82429   3098575  1708010263                 14600 block of HUBBELL ST   
82430   3107616  1708220074                 14600 block of HUBBELL ST   
82431   3160405  1712290074                 9700 block of CHALMER

In [19]:
crime13.head()

ROWNUM   CASEID  CRIMEID        CRNO       CATEGORY  \
0  665662  1629881  1983922  1301310200  MISCELLANEOUS   
1  665663  1618734  1969963  1301010145        LARCENY   
2  665664  1676353  2042431  1305280137  MISCELLANEOUS   
3  665665  1837298  2248340  1407220266          FRAUD   
4  665666  1966686  2416643  1507050157  MISCELLANEOUS   

                     OFFENSEDESCRIPTION  STATEOFFENSEFILECLASS  \
0  MISCELLANEOUS - GENERAL NON-CRIMINAL                99009.0   
1  LARCENY - PURSE SNATCHING - NO FORCE                23002.0   
2     MISCELLANEOUS - ABANDONED VEHICLE                99009.0   
3                         FRAUD (OTHER)                26001.0   
4    MISCELLANEOUS - GENERAL ASSISTANCE                99008.0   

             INCIDENTDATE  HOUR     SCA  PRECINCT                  COUNCIL  \
0  01/01/2013 12:00:00 AM     0  1001.0      10.0  City Council District 2   
1  01/01/2013 12:00:00 AM     0   311.0       3.0  City Council District 6   
2  01/01/2013 12:00:00 AM     0  1111.0      11.0  City Council District 3   
3  01/01/2013 12:00:00 AM     0   210.0       2.0  City Council District 6   
4  01/01/2013 12:00:00 AM     0   203.0       2.0  City Council District 2   

      NEIGHBORHOOD  CENSUSTRACT  \
0     PAUL ROBESON       5303.0   
1              MGM       5207.0   
2     CITY AIRPORT       5106.0   
3          TIREMAN   77772014.0   
4  HARMONY VILLAGE       5367.0   

                                            LOCATION  
0                  14800 HOMLUR\n(42.3989, -83.1364)  
1  01700 THIRD MGM GRAND DETROIT\n(42.3325, -83.0...  
2  11600 MT. ELLIOTT, DETROIT, MI.\n(42.4095, -83...  
3                 06300 CLIFTON\n(42.3487, -83.1339)  
4             15100 MONTE VISTA\n(42.4017, -83.1674)

In [20]:
# lat 40's lon 80's
# forst lat 
crime13['Lat'] = crime13['LOCATION'].map(lambda x: str(x)[20:])
crime13['LOCATION']

0                         14800 HOMLUR\n(42.3989, -83.1364)
1         01700 THIRD MGM GRAND DETROIT\n(42.3325, -83.0...
2         11600 MT. ELLIOTT, DETROIT, MI.\n(42.4095, -83...
3                        06300 CLIFTON\n(42.3487, -83.1339)
4                    15100 MONTE VISTA\n(42.4017, -83.1674)
5                             01300 RD\n(42.3256, -83.0594)
6                         19400 MEYERS\n(42.4359, -83.1705)
7                       13400 MORAN ST\n(42.4157, -83.0629)
8                       03000 LAKEWOOD\n(42.3847, -82.9527)
9               01300 PALLISTER APT 512\n(42.369, -83.0878)
10                     07200 MANSFIELD\n(42.3441, -83.2027)
11                        05700 VERMONT\n(42.357, -83.0877)
12                   09300 N MARTINDALE\n(42.369, -83.1252)
13                            02400 TH\n(42.3365, -83.0616)
14                        22100 MOROSS\n(42.4199, -82.9142)
15                   07700 E JEFFERSON\n(42.3499, -82.9975)
16                 00 NEFF / E. WARREN\n

In [21]:
#lower case
#crime13.NEIGHBORHOOD = crime13.NEIGHBORHOOD.apply(lambda x: x.lower())

In [22]:


#crime13['Location']  = crime13.LOCATION.map(lambda x: str(x)[-18:-1])
crime13['Lat']  = crime13.LOCATION.map(lambda x: str(x)[-18:-11])
crime13['Lon']  = crime13.LOCATION.map(lambda x: str(x)[-10:-1])
#x = crime13['LOCATION']
#crime13['LOCATION'].str.split('( )')
crime13.head()

ROWNUM   CASEID  CRIMEID        CRNO       CATEGORY  \
0  665662  1629881  1983922  1301310200  MISCELLANEOUS   
1  665663  1618734  1969963  1301010145        LARCENY   
2  665664  1676353  2042431  1305280137  MISCELLANEOUS   
3  665665  1837298  2248340  1407220266          FRAUD   
4  665666  1966686  2416643  1507050157  MISCELLANEOUS   

                     OFFENSEDESCRIPTION  STATEOFFENSEFILECLASS  \
0  MISCELLANEOUS - GENERAL NON-CRIMINAL                99009.0   
1  LARCENY - PURSE SNATCHING - NO FORCE                23002.0   
2     MISCELLANEOUS - ABANDONED VEHICLE                99009.0   
3                         FRAUD (OTHER)                26001.0   
4    MISCELLANEOUS - GENERAL ASSISTANCE                99008.0   

             INCIDENTDATE  HOUR     SCA  PRECINCT                  COUNCIL  \
0  01/01/2013 12:00:00 AM     0  1001.0      10.0  City Council District 2   
1  01/01/2013 12:00:00 AM     0   311.0       3.0  City Council District 6   
2  01/01/2013 12:00:00 AM     0  1111.0      11.0  City Council District 3   
3  01/01/2013 12:00:00 AM     0   210.0       2.0  City Council District 6   
4  01/01/2013 12:00:00 AM     0   203.0       2.0  City Council District 2   

      NEIGHBORHOOD  CENSUSTRACT  \
0     PAUL ROBESON       5303.0   
1              MGM       5207.0   
2     CITY AIRPORT       5106.0   
3          TIREMAN   77772014.0   
4  HARMONY VILLAGE       5367.0   

                                            LOCATION      Lat        Lon  
0                  14800 HOMLUR\n(42.3989, -83.1364)  42.3989   -83.1364  
1  01700 THIRD MGM GRAND DETROIT\n(42.3325, -83.0...  42.3325   -83.0578  
2  11600 MT. ELLIOTT, DETROIT, MI.\n(42.4095, -83...  42.4095   -83.0383  
3                 06300 CLIFTON\n(42.3487, -83.1339)  42.3487   -83.1339  
4             15100 MONTE VISTA\n(42.4017, -83.1674)  42.4017   -83.1674

In [23]:
crime13.to_csv('crime13_2018.csv')

In [24]:
# loading the count of points per neigh done one QGIS
crime13count = pd.read_csv('crime2013countGIS.csv')

In [25]:
crime13count['Neighborhood'] = crime13count['new_nhood']
crime13count['Incidents13'] = crime13count['NUMPOINTS']

In [26]:
crime13count

acres    shape_area  join_count  district_nu  objectid_1  \
0    1086.498842  3.171033e+07           3            4           1   
1     496.783595  2.163981e+07           1            4           2   
2    1159.775632  3.118871e+07           1            4           3   
3     901.711919  3.927841e+07           2            4           4   
4     964.174072  4.199925e+07           1            4           5   
5     981.602780  4.275845e+07           1            4           6   
6     857.851272  3.736785e+07           2            3           7   
7     469.754740  2.046243e+07           2            1           8   
8     525.060549  1.409004e+07           1            1           9   
9     382.095201  1.664400e+07           1            1          10   
10    332.194854  1.447035e+07           1            1          11   
11    259.334412  1.129656e+07           1            1          12   
12     95.317515  4.152014e+06           1            1          13   
13    269.390319  1.173460e+07           1            1          14   
14    167.029333  7.275769e+06           1            1          15   
15    376.755572  1.641141e+07           1            1          16   
16     73.899455  3.219047e+06           1            1          17   
17    162.618412  7.083630e+06           1            1          18   
18   1552.524077  6.762768e+07           1            7          19   
19    161.679089  7.042713e+06           1            7          20   
20    813.590587  3.543986e+07           1            7          21   
21    249.731162  1.087825e+07           1            7          22   
22    219.979123  9.582252e+06           2            5          23   
23     53.042219  2.310510e+06           1            5          24   
24    190.175438  8.284009e+06           1            5          25   
25    352.337377  1.534775e+07           1            5          26   
26     90.402275  3.937907e+06           1            5          27   
27   1023.651904  4.459010e+07           2            4          28   
28    417.844224  1.820122e+07           1            6          29   
29    880.430881  3.835142e+07           1            6          30   
..           ...           ...         ...          ...         ...   
178    80.790215  3.519208e+06           1            1         180   
179   391.029984  1.703320e+07           1            1         181   
180   639.899317  2.787390e+07           2            1         182   
181   632.622318  2.755692e+07           1            1         183   
182   234.849353  1.023000e+07           2            1         184   
183   312.922264  1.363084e+07           1            1         185   
184   204.652861  8.914643e+06           1            1         186   
185   928.047138  4.042557e+07           2            1         187   
186    30.722143  1.338251e+06           2            5         188   
187   394.282745  1.717489e+07           1            3         189   
188   180.409455  7.858604e+06           1            3         190   
189   210.914120  9.187382e+06           1            3         191   
190   521.884699  2.273321e+07           1            3         192   
191   344.158439  1.499148e+07           1            3         193   
192   666.965271  2.905289e+07           1            3         194   
193   908.224576  3.956210e+07           2            3         195   
194   307.986427  1.341584e+07           1            3         196   
195   161.591038  7.038877e+06           1            5         197   
196   113.895480  4.961267e+06           1            2         198   
197    75.257550  3.278206e+06           1            2         199   
198   209.687062  9.133932e+06           1            2         200   
199   104.171725  4.537702e+06           1            2         201   
200   153.775629  6.698440e+06           1            2         202   
201   267.018427  1.163128e+07           2            2         203   
202   930.102545  4.051510e+07     

In [27]:
crimecountsmerge = pd.merge(crime13count,  crime17grouped1 , on = 'Neighborhood', how = "inner")

/Users/avigailvantu/anaconda2/envs/myenv/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2961: FutureWarning: 'Neighborhood' is both an index level and a column label.
Defaulting to column, but this will raise an ambiguity error in a future version
  exec(code_obj, self.user_global_ns, self.user_ns)


In [28]:
#here's the merged dataset look like
crimecountsmerge.head()

acres    shape_area  join_count  district_nu  objectid_1  nhood_num  \
0  1086.498842  3.171033e+07           3            4           1         75   
1   496.783595  2.163981e+07           1            4           2         25   
2  1159.775632  3.118871e+07           1            4           3         60   
3   901.711919  3.927841e+07           2            4           4         97   
4   964.174072  4.199925e+07           1            4           5        123   

   objectid    shape_leng          nhood_name           new_nhood  neighborho  \
0         1  27277.074314       Gratiot Woods            West End         103   
1         2  18822.431920       Chandler Park       Chandler Park         105   
2         3  30500.933274           Fox Creek           Riverbend         100   
3         4  25466.579569  Jefferson Chalmers  Jefferson Chalmers         101   
4         5  26596.692068         Morningside         Morningside          97   

   target_fid    shape_le_1  NUMPOINTS        Neighborhood  Incidents13  \
0           1  31380.730047        693            West End          693   
1           2  18822.431920        405       Chandler Park          405   
2           3  22537.847022        675           Riverbend          675   
3           4  25466.579569        654  Jefferson Chalmers          654   
4           5  26596.692068       1514         Morningside         1514   

   Incidents17  
0          352  
1          346  
2          487  
3          468  
4         1102

In [29]:
a = {'Neighborhood': crimecountsmerge['Neighborhood'] , 'Incidents17': crimecountsmerge['Incidents17'], 'Incidents13': crimecountsmerge['Incidents13']}
DetCrimeFinal = pd.DataFrame(data=a)

In [30]:
DetCrimeFinal.head()

Neighborhood  Incidents17  Incidents13
0            West End          352          693
1       Chandler Park          346          405
2           Riverbend          487          675
3  Jefferson Chalmers          468          654
4         Morningside         1102         1514

In [31]:
# export to csv to use in the master notebook 
DetCrimeFinal.to_csv('DetCrime13_17.csv')

## end crime 

## curate 311 data to fix the coordinate format 

In [ ]:
# remove first charc
det_prob['Location'] = det_prob['Location'].map(lambda x: str(x)[10:])

In [ ]:
# remove last charc
det_prob['Location']  = det_prob['Location'] .map(lambda x: str(x)[:-1])

In [ ]:
det_prob['Location'].head(2)

In [ ]:
lat = []
lon = []

# For each row in a varible,
for row in det_prob['Location']:
    # Try to,
    try:
        # Split the row by comma and append
        # everything before the comma to lat
        lat.append(row.split(',')[0])
        # Split the row by comma and append
        # everything after the comma to lon
        lon.append(row.split(',')[1])
    # But if you get an error
    except:
        # append a missing value to lat
        lat.append(np.NaN)
        # append a missing value to lon
        lon.append(np.NaN)

# Create two new columns from lat and lon
det_prob['latitude'] = lat
det_prob['longitude'] = lon


In [ ]:
len(det_prob)

In [ ]:
det_prob['Created At'].min()

In [ ]:
det_prob['Created At'].max()

In [ ]:
#export to csv
det_prob.to_csv('det_prob.csv')


In [ ]:
#new_det = det_prob[(det_prob['Created At'] > '1/1/2017') & (det_prob['Created At']  < '12/31/2017')]
   
   
   
#new_det = det_prob[(det_prob['Created At']>'1/1/2018 00:00:08 AM'and [det_prob['Created At'] <'7/15/2018 00:00:08 AM' ]

In [ ]:
# 02:10:17 PM
det_prob['Creation_year']  = det_prob['Created At'] .map(lambda x: str(x)[:-11])

In [ ]:
det_prob['Creation_year'] = det_prob['Creation_year'].map(lambda x: str(x)[6:])

In [ ]:
det_prob[2000:2002]

In [ ]:
det_prob.Creation_year.unique()

In [ ]:
new_det = det_prob[det_prob['Creation_year']== '2018 ']

In [ ]:
len(new_det)

## export 311 data for viz

In [ ]:
new_det.to_csv('det_2018.csv')